CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs

```diff
@@ -128,20 +183,65 @@ private static async Task<Operation> MonitorOperation(IQnAMakerClient qnaClient,
                 await Task.Delay(5000);
                 operation = await qnaClient.Operations.GetDetailsAsync(operation.OperationId);
             }
+            return operation;
+        }
+        // </MonitorOperation>
 
-            if (operation.OperationState != OperationStateType.Succeeded)
+        // Checks valid file type before sending for extraction
+        private static bool IsValidFile(string fileName)
+        {
+            HashSet<string> fileExtensions = new HashSet<string> { "tsv", "pdf", "txt", "docx", "xlsx" };
+            string urlExtension;
+            try
             {
-                log.LogError($"Operation {operation.OperationId} failed to completed. ErrorMessage: {operation.ErrorResponse.Error.Message}");
+                urlExtension = Path.GetExtension(fileName).ToLower().TrimStart('.');
             }
-            return operation;
+            catch
+            {
+                urlExtension = string.Empty;
+            }
+            
+            return fileExtensions.Contains(urlExtension);
+        }
+
+        private static string GetOperationStatus(Operation operation, string fileName, ILogger log)
+        {
+            string operationState = OperationStateType.Succeeded;
+            if (operation.OperationState != OperationStateType.Succeeded && operation.ErrorResponse != null)
+            {
+                var error = operation.ErrorResponse.Error.Details.Where(error => error.Target == fileName);
+                if(error != null && error.Any())
+                {
+                    var errorResponse = error.ToList().First();
```

Comment:
Add another comment here to explain, as we discussed

Reply:
done